Core/contentblocker.js

```diff
@@ -62,9 +62,9 @@ var duckduckgoContentBlocking = function() {
 
 	// private 
 	function getTopLevelURL() {
-		// can throw a security exception if called from a frame with document loaded from different url to top (mainly when blocking is disabled)
 		try {
-			return new URL(top.location.href)
+			// FROM: https://stackoverflow.com/a/7739035/73479		
+			return new URL(window.location != window.parent.location ? document.referrer : document.location.href)
```

Comment:
This makes sense, although I'm concerned that there may be edge cases where document.referrer does not give you the top level document. I'm not a javascript expert, let's get someone who is to double-check. Maybe @nilnilnil?

Reply:
I'd like to see another comment about the why. I think it's OK, or something to try.

Reply:
Just something to throw into the hat, there *could* be some trial and error here with regard to how the referrer policy is being set across services/sites: https://w3c.github.io/webappsec-referrer-policy/#referrer-policies i.e. there may be cases where document.referrer is undefined, but I guess that would be no different than what's in place without this fix?